front-end/src/components/User.js

```diff
@@ -54,6 +56,19 @@ const User = (props) => {
     getAllSecrets();
   }, [secretUpdate]);
 
+  const deleteSecret = (secretId) => {
+    axios
+      .delete(`${process.env.REACT_APP_BACKEND_SERVICE}/delete-secret/${secretId}`)
+      .then((res) => {
+        if (res.status === 200) {
+          getAllSecrets(); // re-render the page
```

Comment:
instead of making a `getAllSecrets()` call here you can do `setSecretUpdate(secretUpdate - 1)` and that will automatically do the getAllSecrets() call and reset the secrets count here.

Reply:
why do this instead of calling getAllSecrets directly?

Reply:
`secretUpdate` is a counter setup to see how many secrets there are for that user, so it's just good housekeeping to update that counter. Every time it is updated it get's called in `useEffect` to re-render the data.